File: solutions/cpp/209/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0, curr = 0, ans = INT_MAX;
        for (int right = 0; right < nums.size(); right++) {
            curr += nums[right];
            while (curr >= target) {
                ans = min(ans, right - left + 1);
                curr -= nums[left];
                left++;
            }
        }

        if (ans == INT_MAX) {
            return 0;
        }
        return ans;
    }
};

File: solutions/golang/209/solution.go
func minSubArrayLen(target int, nums []int) int {
	left := 0
	ans := math.MaxInt
	curr := 0
	for right := 0; right < len(nums); right++ {
		curr += nums[right]
		for curr >= target {
			ans = min(ans, right-left+1)
			curr -= nums[left]
			left++
		}
	}

	if ans == math.MaxInt {
		return 0
	}

	return ans
}